### DISCUSSION TITLE
remove declaration:  using Uri = ::winrt::Windows::Foundation::Uri;

### DISCUSSION THREAD

**Discussion by shelllet** [microsoft/microsoft-ui-xaml#10224] [General]
![image](https://github.com/user-attachments/assets/73aa42b5-d44f-4011-8ef6-203e3611afa5)

why add ` using Uri = ::winrt::Windows::Foundation::Uri;` in line 39 

this could easily conflict whith my definition class Uri.

![image](https://github.com/user-attachments/assets/0624981e-c3e0-4bae-a227-5c4fcb12024e)

**Comment by DarranRowe** [comment]
The thing to remember is that using is a typedef.

To provide a short and silly example, including names that are hard to confuse with each other:

```
namespace a
{
	struct my_struct
	{
		unsigned int blerrrr;
	};
}

namespace b
{
	struct my_struct
	{
		float snmarra;
	};
}

int wmain()
{
	using my_struct = a::my_struct;

	b::my_struct bms{};

	bms.snmarra = 1.f;

	return 0;
}
```

Using a fully qualified name overrides the typedef, it is only if the name is unqualified that any type definitions are taken into account. What's more, C++ has interesting scoping rules.

```
namespace a
{
	struct my_struct
	{
		unsigned int blerrrr;
	};
}

namespace b
{
	struct my_struct
	{
		float snmarra;
	};
}

int wmain()
{
	using my_struct = a::my_struct;

	{
		using my_struct = b::my_struct;

		my_struct bms{};

		bms.snmarra = 1.f;
	}

	return 0;
}
```

This means that the type definition can always be overridden at any scope.

Finally, to extend the code around that set of using statements up a little:

```
namespace winrt::TestMeh::implementation
{


    using Application = ::winrt::Microsoft::UI::Xaml::Application;
    using ComponentResourceLocation = ::winrt::Microsoft::UI::Xaml::Controls::Primitives::ComponentResourceLocation;
    using DataTemplate = ::winrt::Microsoft::UI::Xaml::DataTemplate;
    using DependencyObject = ::winrt::Microsoft::UI::Xaml::DependencyObject;
    using DependencyProperty = ::winrt::Microsoft::UI::Xaml::DependencyProperty;
    using IComponentConnector = ::winrt::Microsoft::UI::Xaml::Markup::IComponentConnector;
    using Uri = ::winrt::Windows::Foundation::Uri;
    using XamlBindingHelper = ::winrt::Microsoft::UI::Xaml::Markup::XamlBindingHelper;

```

The type definition is in a namespace. Because of the points I made, the only way this could be causing issues is if you are using a typedef or the using statement to remove your Uri namespace at a global level, and the generated implementation is overriding you in the component's namespace. Doing this kind of thing at global scope, especially in headers, tends to be seen as bad practice. If you want to reduce typing, then either use a namespace alias, or do this in a .cpp file at a more restrictive level.
